x-pack/plugins/ml/public/components/rule_editor/rule_editor_flyout.test.js

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+// Mock the services required for reading and writing job data.
+jest.mock('../../services/job_service', () => ({
+  mlJobService: {
+    getJob: () => {
+      return {
```

Comment:
total nit but this return object could be a separate JSON file which we import. resonates with the idea James' mentioned that something like a mock for `mlJobService` could be made reusable for other tests.

Reply:
Yes, as some of these mocked functions start needing to be reused I agree it would be a good idea to move things like test job configs into separate files. We'd need a wide range of job configs to cover all the things we're testing - here for example I want a detector with no partitioning fields and a single rule.